docs/dev/plugins.md

```diff
@@ -30,6 +30,12 @@ The normal distribution and installation method is via gems, handled by the `ins
 
 For more on the `plugin` CLI command, run `inspec plugin help`.
 
+### Plugins can be bundled with a profile
+
+If plugin code is tightly coupled to one profile only, you can bundle it into the profile by placing it into the `libraries/` directory.
+
+The plugin name must be prefixed with "profile" to handle loading properly.
```

Comment:
It would be good to give an example of what a plugin inside `libraries` looks like, that fixture could be linked to (or repeated) here.

We should agree on `^profile.*` meaning a profile bundled plugin (i.e. we cannot start the name with `profile` for a plugin not bundled with the profile, or for other code like a custom resource that is bundled in `libraries`)